src/hooks/engine.rs

```diff
@@ -973,6 +1003,40 @@ pub mod exported {
         })
     }
 
+    #[cfg_attr(
+        not(feature = "bxt-compatibility"),
+        export_name = "R_DrawSequentialPoly"
+    )]
+    pub unsafe extern "C" fn my_R_DrawSequentialPoly(
+        surf: *mut c_void,
+        face: *mut c_int,
+    ) -> *mut c_void {
+        abort_on_panic(move || {
+            let marker = MainThreadMarker::new();
+
+            if wallhack::is_active(marker) {
```

Comment:
I think it'll be cleaner if the `wallhack::is_active()` check is inside `with_wallhack` and `with_after_wallhack` so there's no repetition here at the usage site.

Reply:
Like this? https://github.com/YaLTeR/bxt-rs/pull/15/commits/6fe2f37108a630ca5ac2ac03700386c0e8fde26f

Reply:
Yes exactly